Add "Global State" concept (#980)
casper-network/docs: source/docs/casper/concepts/global-state.md

# Global State {#global-state-head}

## Introduction {#global-state-intro}

> The "global state" is the storage layer for the blockchain. All accounts, contracts, and any associated data they have are stored in the global state. Our global state has the semantics of a key-value store (with additional permissions logic since not all users can access all values in the same way).

Reviewer: ```suggestion
"Global state" is the storage layer for the blockchain. Storage of all accounts, contracts, and their associated data occurs in global state. Our global state has the semantics of a key-value store (with additional permissions logic since not all users can access all values in the same way).
```
